### PR TITLE
Tidy up validation for storage/zfs modules

### PR DESCRIPTION
**Pull request by russoz**
##### SUMMARY
Tidy up validation for storage/zfs modules

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
plugins/modules/storage/zfs/zpool_facts.py
plugins/modules/storage/zfs/zfs_facts.py
plugins/modules/storage/zfs/zfs_delegate_admin.py
plugins/modules/storage/zfs/zfs.py

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Looks good. The change to zfs_delegate_admin requires a changelog fragment (I think the module was crashing before when non-strings were provided in these lists).

**Comment by russoz** [comment]
> Looks good. The change to zfs_delegate_admin requires a changelog fragment (I think the module was crashing before when non-strings were provided in these lists).

Done

File: changelogs/fragments/1766-zfs-fixed-sanity.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - zfs_delegate_admin - added ``elements`` to documentation and ``argument_spec`` (https://github.com/ansible-collections/community.general/pull/1766).
```
Comment (felixfontein) [inline changelogs/fragments/1766-zfs-fixed-sanity.yml:2]:
```suggestion
  - zfs_delegate_admin - the elements of ``users``, ``groups`` and ``permissions`` are now enforced to be strings (https://github.com/ansible-collections/community.general/pull/1766).
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
I'll keep this open for some time (before the next release, probably mid next week or so) so maintainers can take a look as well.

**Comment by felixfontein** [comment]
Update: I will merge this already on Monday so @russoz can proceed with more cleanup PRs.

**Comment by felixfontein** [comment]
@russoz thanks for fixing these!

**Pull request merged**